### PR TITLE
python311Packages.mitmproxy-macos: 0.4.1 -> 0.5.1, python311Packages.mitmproxy-rs: 0.4.1 -> 0.5.1 

### PR DESCRIPTION
**Pull request by fabaff**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 280050` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>mitmproxy (python311Packages.mitmproxy)</li>
    <li>mitmproxy.dist (python311Packages.mitmproxy.dist)</li>
    <li>mitmproxy2swagger</li>
    <li>mitmproxy2swagger.dist</li>
    <li>python310Packages.mitmproxy</li>
    <li>python310Packages.mitmproxy-rs</li>
    <li>python310Packages.mitmproxy-rs.dist</li>
    <li>python310Packages.mitmproxy.dist</li>
    <li>python311Packages.mitmproxy-rs</li>
    <li>python311Packages.mitmproxy-rs.dist</li>
    <li>wapiti</li>
    <li>wapiti.dist</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 280050` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>mitmproxy (python311Packages.mitmproxy)</li>
    <li>mitmproxy.dist (python311Packages.mitmproxy.dist)</li>
    <li>mitmproxy2swagger</li>
    <li>mitmproxy2swagger.dist</li>
    <li>python310Packages.mitmproxy</li>
    <li>python310Packages.mitmproxy-macos</li>
    <li>python310Packages.mitmproxy-macos.dist</li>
    <li>python310Packages.mitmproxy-rs</li>
    <li>python310Packages.mitmproxy-rs.dist</li>
    <li>python310Packages.mitmproxy.dist</li>
    <li>python311Packages.mitmproxy-macos</li>
    <li>python311Packages.mitmproxy-macos.dist</li>
    <li>python311Packages.mitmproxy-rs</li>
    <li>python311Packages.mitmproxy-rs.dist</li>
    <li>wapiti</li>
    <li>wapiti.dist</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 280050` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>mitmproxy (python311Packages.mitmproxy)</li>
    <li>mitmproxy.dist (python311Packages.mitmproxy.dist)</li>
    <li>mitmproxy2swagger</li>
    <li>mitmproxy2swagger.dist</li>
    <li>python310Packages.mitmproxy</li>
    <li>python310Packages.mitmproxy-rs</li>
    <li>python310Packages.mitmproxy-rs.dist</li>
    <li>python310Packages.mitmproxy.dist</li>
    <li>python311Packages.mitmproxy-rs</li>
    <li>python311Packages.mitmproxy-rs.dist</li>
    <li>wapiti</li>
    <li>wapiti.dist</li>
  </ul>
</details>

**Review by boltzmannrain — APPROVED** [review]

**Review by risicle — APPROVED** [review]
LGTM, `nixpkgs-review` happy, macos 12 x86_64, nixos x86_64

File: pkgs/development/python-modules/mitmproxy-rs/default.nix [hunk 1]
```diff
@@ -10,21 +10,18 @@
...
   src = fetchFromGitHub {
     owner = "mitmproxy";
     repo = "mitmproxy_rs";
     rev = version;
-    hash = "sha256-Vc7ez/W40CefO2ZLAHot14p478pDPtQor865675vCtI=";
+    hash = "sha256-nrm1T2yaGVmYsubwNJHPnPDC/A/jYiKVzwBKmuc9MD4=";
   };
 
   cargoDeps = rustPlatform.importCargoLock {
     lockFile = ./Cargo.lock;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mitmproxy-rs/default.nix:24]: If there are not git dependencies, we don't need to vendor the lockf file.

**Pull request merged**